internal/interceptors.go

```diff
@@ -86,6 +87,8 @@ type WorkflowOutboundCallsInterceptor interface {
 	IsReplaying(ctx Context) bool
 	HasLastCompletionResult(ctx Context) bool
 	GetLastCompletionResult(ctx Context, d ...interface{}) error
+	HasLastFailure(ctx Context) bool
```

Comment:
I don't think HasLastFailure is needed. We can just return nil from GetLastFailure. It didn't work for the completion result as it required deserialization and didn't return anything.